ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/transaction/arjunacore/jca/TransactionImporterImple.java

```diff
@@ -211,17 +238,33 @@ public void removeImportedTransaction(Xid xid) throws XAException
 	}
 	
 	public Set<Xid> getInflightXids(String parentNodeName) {
-		Iterator<TransactionImple> iterator = _transactions.values().iterator();
+		Iterator<TransactionImpleHolder> iterator = _transactions.values().iterator();
 		Set<Xid> toReturn = new HashSet<Xid>();
 		while (iterator.hasNext()) {
-			TransactionImple next = iterator.next();
-			if (next.getParentNodeName().equals(parentNodeName)) {
-				toReturn.add(next.baseXid());
+			TransactionImpleHolder next = iterator.next();
+			TransactionImple imported = null;
+			try {
+				imported = next.getImported();
+			} catch (InterruptedException e) {
+				throw new RuntimeException(e);
```

Comment:
The contract for importTransaction (and recoverTransaction) is to either return a valid TransactionImple or to throw an XAException. Therefore if the wait() call is interrupted the only option is to wrap the InterruptedException as either a RuntimeException of an XA_RETRY exception.